storage/client.go

```diff
@@ -308,6 +308,15 @@ func (c Client) buildCanonicalizedResourceTable(uri string) (string, error) {
 		cr += u.EscapedPath()
 	}
 
+	params, err := url.ParseQuery(u.RawQuery)
+
+	// search for "comp" parameter, if exists then add it to canonicalizedresource
+	for key := range params {
```

Comment:
From [Authentication for the Azure Storage Services](https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/authentication-for-the-azure-storage-services) (the "Constructing the Canonicalized Resource String" section, I can guess it has to do with that...

> Any portion of the CanonicalizedResource string that is derived from the resource's URI should be encoded exactly as it is in the URI.

Still asking the Storage team to understand this better.